### ISSUE TITLE
Need option to opt-out dev container features

### ISSUE THREAD

**Issue by developStorm** [microsoft/vscode-remote-release#7378]
Right now the extension always builds `vsc-workspace-id-features` image on top of the images specified in `devcontainer.json`, even when no dev container feature is used. 

If user specifies `--pull always` in `runArgs`, this behavior causes problems: Docker will try to pull the image from docker hub (and would't find it), despite it was just built locally by the extension, causing the workspace to not load. It also breaks pulling policy specified by user as now the local image is used so there's nothing to pull unless a rebuild is manually triggered on updated image.

I can think of two fixes of this for now: disable this local feature image creation if there's no dev container feature in used, or add an option to allow users explicitly opt-out from this local image creation.

Thanks for helping out!

- VSCode Version: 1.72.2
- Reproduces in: Remote - Containers 

Steps to Reproduce:

1. create an `devcontainer.json` with `"image":` and `runArgs: ["--pull", "always"]`
2. open the workspace with Remote - Containers.
3. the workspace failed to load with docker error "...requested access to the resource is denied...", as it tries to pull a non-exists image from hub.

**Comment by mtaron** [comment]
Just ran into this as well. [Prebuilding](https://code.visualstudio.com/docs/remote/containers#_prebuilding-dev-container-images) devcontainers goes hand in hand with `--pull=always` and but the features feature has broken that workflow.

**Comment by chrmarti** [comment]
We are adding metadata to the image now (in addition to features). There might be cases where that is not needed or can be avoided.

For the prebuild case: There are `Dev Containers: Rebuild Without Cache and Reopen in Container` and `Dev Containers: Rebuild Without Cache` commands that will pull newer versions. Does that work?

**Comment by developStorm** [comment]
@chrmarti Thanks for the info! Rebuild manually is feasible, but it requires more human interventions which might not work well in some use cases. For example, we currently distubute dev container configurations to our students in entry level algorithm course to help them with enviorment setup, but we apprently wouldn't want to expose the whole complextiy of Docker and dev containers to those freshman. When there's an image update, the current approach requires us to instruct all students to maually rebuild without cache - when they [probably don't even know what's cache 😂](https://www.reddit.com/r/uofm/comments/y6l9x1/cash/). 

I know this is a corner case, but it would be still desired to run workloads directly on prebuilt images. Appriciate your consideration!

**Comment by mtaron** [comment]
I just checked and `Dev Containers: Rebuild Without Cache and Reopen in Container` does _not_ pull automatically.

Here are my code/extension versions if that's helpful.

```shell
❯ code --version
1.72.2
d045a5eda657f4d7b676dedbfa7aab8207f8a075
x64
```

```shell
❯ code --list-extensions --show-versions
eamodio.gitlens@13.0.3
GitHub.github-vscode-theme@6.3.2
hashicorp.terraform@2.24.3
mads-hartmann.bash-ide-vscode@1.14.0
ms-azuretools.vscode-docker@1.22.2
ms-kubernetes-tools.vscode-kubernetes-tools@1.3.10
ms-toolsai.jupyter-keymap@1.0.0
ms-vscode-remote.remote-containers@0.255.4
redhat.vscode-yaml@1.10.1
tamasfe.even-better-toml@0.18.3
Tim-Koehler.helm-intellisense@0.14.3
timonwong.shellcheck@0.26.2
```

**Comment by chrmarti** [comment]
@mtaron This was recently reported when using a devcontainer.json with `"image"`. Will continue there.

@developStorm Would it be an option to use `docker pull` in the `"initializeCommand"`?

**Comment by developStorm** [comment]
@chrmarti In the specific use case, not really because we run `docker image prune -f` in `initializeCommand` for users (to avoid over using their disk space). If dev container doesn't directly run on the image our base image will be pruned if I understand the prune mechanism correctly ; (. You can see the full configuration [here](https://github.com/CAENTainer/EECS281-Devcontainer/blob/f844bc53f8f7f84f1bb516d753c8d8275093dae8/Scaffold/.devcontainer/devcontainer.json#L13-L21)

Again it's our very special use case intended for "non/new-technical" users, but still, if there's no particular reason that we have to keep metadata for image with no feature, why we don't just remove that extra local layer?

**Comment by chrmarti** [comment]
This fix is available in Dev Containers 0.264.0-pre-release. Let me know if that fixes this issue for you. Thanks.

**Comment by developStorm** [comment]
@chrmarti can confirm the issue is fixed in pre release now. Thank you!
